src/AVSRegistrar.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import {IAVSRegistrar} from "eigenlayer-contracts/src/contracts/interfaces/IAVSRegistrar.sol";
+import {IRegistryCoordinator} from "./interfaces/IRegistryCoordinator.sol";
+import {IBLSApkRegistry} from "./interfaces/IBLSApkRegistry.sol";
+import {IStakeRegistry} from "./interfaces/IStakeRegistry.sol";
+import {IIndexRegistry} from "./interfaces/IIndexRegistry.sol";
+
+abstract contract AVSRegistrar is IAVSRegistrar {
```

Comment:
I was initially thinking it would be useful to encapsulate the logic here but realized all the logic would still be in the RegistryCoordinator due to the amount of state / coordinator required to change on these hooks